src/se/vidstige/jadb/managers/Intent.java

```diff
@@ -0,0 +1,761 @@
+            flags.add("--grant-read-uri-permission");
+        }
+        if((mFlags & FLAG_GRANT_WRITE_URI_PERMISSION) != 0) {
+            flags.add("--grant-write-uri-permission");
+        }
+        if((mFlags & FLAG_GRANT_PERSISTABLE_URI_PERMISSION) != 0) {
+            flags.add("--grant-persistable-uri-permission");
+        }
+        if((mFlags & FLAG_GRANT_PREFIX_URI_PERMISSION) != 0) {
+            flags.add("--grant-prefix-uri-permission");
+        }
+        if((mFlags & FLAG_EXCLUDE_STOPPED_PACKAGES) != 0) {
+            flags.add("--exclude-stopped-packages");
+        }
+        if((mFlags & FLAG_INCLUDE_STOPPED_PACKAGES) != 0) {
+            flags.add("--include-stopped-packages");
+        }
+        if((mFlags & FLAG_DEBUG_LOG_RESOLUTION) != 0) {
+            flags.add("--debug-log-resolution");
+        }
+        if((mFlags & FLAG_ACTIVITY_BROUGHT_TO_FRONT) != 0) {
+            flags.add("--activity-brought-to-front");
+        }
+        if((mFlags & FLAG_ACTIVITY_CLEAR_TOP) != 0) {
+            flags.add("--activity-clear-top");
+        }
+        if((mFlags & FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET) != 0) {
+            flags.add("--activity-clear-when-task-reset");
+        }
+        if((mFlags & FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS) != 0) {
+            flags.add("--activity-exclude-from-recents");
+        }
+        if((mFlags & FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY) != 0) {
+            flags.add("--activity-launched-from-history");
+        }
+        if((mFlags & FLAG_ACTIVITY_MULTIPLE_TASK) != 0) {
+            flags.add("--activity-multiple-task");
+        }
+        if((mFlags & FLAG_ACTIVITY_NO_ANIMATION) != 0) {
+            flags.add("--activity-no-animation");
+        }
+        if((mFlags & FLAG_ACTIVITY_NO_HISTORY) != 0) {
+            flags.add("--activity-no-history");
+        }
+        if((mFlags & FLAG_ACTIVITY_NO_USER_ACTION) != 0) {
+            flags.add("--activity-no-user-action");
+        }
+        if((mFlags & FLAG_ACTIVITY_PREVIOUS_IS_TOP) != 0) {
+            flags.add("--activity-previous-is-top");
+        }
+        if((mFlags & FLAG_ACTIVITY_REORDER_TO_FRONT) != 0) {
+            flags.add("--activity-reorder-to-front");
+        }
+        if((mFlags & FLAG_ACTIVITY_RESET_TASK_IF_NEEDED) != 0) {
+            flags.add("--activity-reset-task-if-needed");
+        }
+        if((mFlags & FLAG_ACTIVITY_SINGLE_TOP) != 0) {
+            flags.add("--activity-single-top");
+        }
+        if((mFlags & FLAG_ACTIVITY_CLEAR_TASK) != 0) {
+            flags.add("--activity-clear-task");
+        }
+        if((mFlags & FLAG_ACTIVITY_TASK_ON_HOME) != 0) {
+            flags.add("--activity-task-on-home");
+        }
+        if((mFlags & FLAG_RECEIVER_REGISTERED_ONLY) != 0) {
+            flags.add("--receiver-registered-only");
+        }
+        if((mFlags & FLAG_RECEIVER_REPLACE_PENDING) != 0) {
+            flags.add("--receiver-replace-pending");
+        }
+    }
+}
```

Comment:
The `TODO` here are in case the documentation should be kept. All documentation is from the source file.